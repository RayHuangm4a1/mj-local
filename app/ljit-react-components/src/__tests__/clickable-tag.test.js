import React from 'react';
import { shallow, mount, } from 'enzyme';
import { MemoryRouter as Router, } from 'react-router-dom';
import ClickableTag from '../components/clickable-tag';

const { ColorEnums, BadgeAlignmentEnums, } = ClickableTag;

describe('ClickableTag', () => {
	it ('should handle default props', () => {
		const {
			backgroundColor,
			textColor,
			text,
			color,
			badgeAlignment,
			hasBadge,
			badgeIcon,
			onClick,
		} = ClickableTag.defaultProps;

		expect(backgroundColor).toEqual('');
		expect(textColor).toEqual('');
		expect(text).toEqual('');
		expect(color).toEqual(ColorEnums.BLACK);
		expect(badgeAlignment).toEqual(BadgeAlignmentEnums.LEFT);
		expect(hasBadge).toEqual(false);
		expect(badgeIcon).toEqual(0);
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it ('should render correctly', () => {
		const backgroundColor = 'blue';
		const textColor = 'yellow';
		const text = 'mock-text';
		const className = 'mock-class';
		const style = { fontSize: '20px', };
		const color = ColorEnums.BLACK;
		const badgeAlignment = BadgeAlignmentEnums.LEFT;
		const badgeIcon = 0;
		const hasBadge = true;
		const onClick = () => {};
		const wrapper = shallow(
			<ClickableTag
				backgroundColor={backgroundColor}
				textColor={textColor}
				text={text}
				color={color}
				className={className}
				badgeAlignment={badgeAlignment}
				badgeIcon={badgeIcon}
				hasBadge={hasBadge}
				onClick={onClick}
				style={style}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it ('should be selectable by class ljit-clickable-tag', () => {
		const wrapper = shallow(
			<ClickableTag text={'mock-text'} />
		);

		expect(wrapper.hasClass('ljit-clickable-tag')).toEqual(true);
	});

	it ('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<ClickableTag text={'mock-text'} className={className}/>
		);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it ('should be selectable by class ljit-clickable-tag--with-badge when hasBadge passed as true', () => {
		const wrapper = shallow(
			<ClickableTag text={'mock-text'} hasBadge={true} />
		);

		expect(wrapper.hasClass('ljit-clickable-tag--with-badge')).toEqual(true);
	});

	it ('should be selectable by class ljit-clickable-tag--with-left-badge when badgeAlignment passed as left', () => {
		const wrapper = shallow(
			<ClickableTag text={'mock-text'} hasBadge={true} badgeAlignment={BadgeAlignmentEnums.LEFT} />
		);

		expect(wrapper.hasClass('ljit-clickable-tag--with-left-badge')).toEqual(true);
	});

	it ('should be selectable by class ljit-clickable-tag--with-right-badge when badgeAlignment passed as left', () => {
		const wrapper = shallow(
			<ClickableTag text={'mock-text'} hasBadge={true} badgeAlignment={BadgeAlignmentEnums.RIGHT} />
		);

		expect(wrapper.hasClass('ljit-clickable-tag--with-right-badge')).toEqual(true);
	});

	it ('should mount in a full DOM', () => {
		const backgroundColor = 'blue';
		const textColor = 'yellow';
		const text = 'mock-text';
		const className = 'mock-class';
		const style = { fontSize: '20px', };
		const color = ColorEnums.BLACK;
		const badgeAlignment = BadgeAlignmentEnums.LEFT;
		const badgeIcon = 0;
		const hasBadge = true;
		const onClick = () => {};
		const wrapper = mount(
			<Router>
				<ClickableTag
					backgroundColor={backgroundColor}
					textColor={textColor}
					text={text}
					style={style}
					className={className}
					color={color}
					badgeAlignment={badgeAlignment}
					badgeIcon={badgeIcon}
					hasBadge={hasBadge}
					onClick={onClick}
				/>
			</Router>
		);

		expect(wrapper.find('ClickableTag').props().backgroundColor).toEqual(backgroundColor);
		expect(wrapper.find('ClickableTag').props().textColor).toEqual(textColor);
		expect(wrapper.find('ClickableTag').props().text).toEqual(text);
		expect(wrapper.find('ClickableTag').props().className).toEqual(className);
		expect(wrapper.find('ClickableTag').props().style).toMatchObject(style);
		expect(wrapper.find('ClickableTag').props().color).toEqual(color);
		expect(wrapper.find('ClickableTag').props().badgeAlignment).toEqual(badgeAlignment);
		expect(wrapper.find('ClickableTag').props().badgeIcon).toEqual(badgeIcon);
		expect(wrapper.find('ClickableTag').props().hasBadge).toEqual(hasBadge);
		expect(wrapper.find('ClickableTag').props().onClick).toEqual(onClick);
	});
});

describe('LinkTag BadgeAlignment Enums', () => {
	it('should contain LEFT property', () => {
		const typeName = 'left';
		const formatType = 'LEFT';

		expect(BadgeAlignmentEnums[formatType]).toEqual(typeName);
	});

	it('should contain RIGHT property', () => {
		const typeName = 'right';
		const formatType = 'RIGHT';

		expect(BadgeAlignmentEnums[formatType]).toEqual(typeName);
	});
});