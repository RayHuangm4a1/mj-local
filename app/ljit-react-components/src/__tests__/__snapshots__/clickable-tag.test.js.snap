// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableTag should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClickableTag
    backgroundColor="blue"
    badgeAlignment="left"
    badgeIcon={0}
    className="mock-class"
    color="black"
    hasBadge={true}
    onClick={[Function]}
    style={
      Object {
        "fontSize": "20px",
      }
    }
    text="mock-text"
    textColor="yellow"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="ljit-clickable-tag mock-class"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        <Tag
          backgroundColor="blue"
          color="black"
          hasBorder={false}
          isClosable={false}
          onClose={[Function]}
          shape="rectangle"
          text="mock-text"
          textColor="yellow"
        />
      </div>,
      "className": "ljit-clickable-tag--with-badge ljit-clickable-tag--with-left-badge",
      "count": 0,
      "isBordered": true,
      "isZeroVisible": true,
      "overflowCount": 99,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Tag
          backgroundColor="blue"
          color="black"
          hasBorder={false}
          isClosable={false}
          onClose={[Function]}
          shape="rectangle"
          text="mock-text"
          textColor="yellow"
        />,
        "className": "ljit-clickable-tag mock-class",
        "onClick": [Function],
        "style": Object {
          "fontSize": "20px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": "blue",
          "color": "black",
          "hasBorder": false,
          "isClosable": false,
          "onClose": [Function],
          "shape": "rectangle",
          "text": "mock-text",
          "textColor": "yellow",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="ljit-clickable-tag mock-class"
          onClick={[Function]}
          style={
            Object {
              "fontSize": "20px",
            }
          }
        >
          <Tag
            backgroundColor="blue"
            color="black"
            hasBorder={false}
            isClosable={false}
            onClose={[Function]}
            shape="rectangle"
            text="mock-text"
            textColor="yellow"
          />
        </div>,
        "className": "ljit-clickable-tag--with-badge ljit-clickable-tag--with-left-badge",
        "count": 0,
        "isBordered": true,
        "isZeroVisible": true,
        "overflowCount": 99,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Tag
            backgroundColor="blue"
            color="black"
            hasBorder={false}
            isClosable={false}
            onClose={[Function]}
            shape="rectangle"
            text="mock-text"
            textColor="yellow"
          />,
          "className": "ljit-clickable-tag mock-class",
          "onClick": [Function],
          "style": Object {
            "fontSize": "20px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": "blue",
            "color": "black",
            "hasBorder": false,
            "isClosable": false,
            "onClose": [Function],
            "shape": "rectangle",
            "text": "mock-text",
            "textColor": "yellow",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
